### PR TITLE
Support custom `Error` subclass in the TypeScript types

### PR DESCRIPTION
**Pull request by gr2m**
First of all thanks for the great library! We use it in `semantic-release` and love it!

I'm now looking into using it in [`@octokit/webhooks`](https://github.com/octokit/webhooks) in order to address https://github.com/octokit/webhooks.js/issues/182

We use a custom Error type for request errors, which are the most common error types in webhook event handlers, so I'd like to provide better intellisense when users interact with the errors from the AggregatedError array. In order to make that possible, the Error type used by would need to be customizeable. That's what I do in this pull request.

Sorry for not opening an issue first, I figured it would be easier to reason about the implementation details in a pull request

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -1,8 +1,9 @@
 /**
 Create an error from multiple errors.
 */
-declare class AggregateError extends Error implements Iterable<Error> {
-	readonly name: 'AggregateError';
+declare class AggregateError<T extends Error = Error> extends Error
+	implements Iterable<T> {
+	readonly name: "AggregateError";
```
Comment (sindresorhus) [inline index.d.ts]: Don't do unrelated changes

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 1 above)
Comment (gr2m) [reply index.d.ts]: sorry that I missed that, I'll take care of it right away

**Comment by gr2m** [comment]
> Don't do unrelated changes

fixed

File: index.test-d.ts [hunk 2]
```diff
@@ -12,3 +12,14 @@ expectType<IterableIterator<Error>>(aggregateError[Symbol.iterator]());
 for (const error of aggregateError) {
 	expectType<Error>(error);
 }
+
+interface CustomError extends Error {
+	foo: string;
+}
+const customAggregateError = new AggregateError<CustomError>([
+	new Error('foo')
```
Comment (sindresorhus) [inline index.test-d.ts]: Can you make an actual Error subclass and use it here instead?

Comment (gr2m) [reply index.test-d.ts]: like this 6d23959?

Comment (sindresorhus) [reply index.test-d.ts]: Yes
